### ISSUE TITLE
Unable to log back in

### ISSUE THREAD

**Issue by jasonpearce** [ManufacturaInd/shaarli-2004licious-theme#2]
Thanks for the retro del.icio.us theme. Works well until you log out. When I click on the login link (https://bookmarks.jasonpearce.com/?do=login, for me), the page doesn't render and has PHP errors. 

I'm using v0.8.4 of Shaarli. Had to just switch back to the default theme so I could log back in. Now that del.iciou.us is dead, it would be nice to have your team as the ideal replacement.

**Comment by rlafuente** [comment]
Hey Jason, thanks for this issue. We didn't in fact finish our work in converting all Shaarli templates into this theme, and the log in page is one of those. 

We're now going back to this to see if we can fix that quickly -- we're super grateful for your kind words and want to help you get it up and running.

**Comment by xuv** [comment]
Indeed, just tried with the latest shaarli and the 2004licious theme and it does not render any page for me. Will see if I can see where the error is coming from.

**Comment by xuv** [comment]
This is the PHP error I get:
```
Fatal error: Uncaught exception 'RainTpl_NotFoundException' with message 'Template linklist not found!' in shaarli/inc/rain.tpl.class.php:268 Stack trace: 
#0 shaarli/inc/rain.tpl.class.php(164): RainTPL->check_template('linklist') 
#1 shaarli/application/PageBuilder.php(130): RainTPL->draw('linklist') 
#2 shaarli/index.php(758): PageBuilder->renderPage('linklist') 
#3 shaarli/index.php(1083): showLinkList(Object(PageBuilder), Object(LinkDB)) 
#4 shaarli/index.php(2258): renderPage() 
#5 {main} thrown in shaarli/inc/rain.tpl.class.php on line 268
```

**Comment by xuv** [comment]
Ooops. Sorry. Ignore that last error I posted. It was related to a trailing slash missing from the config file. I've now updated the theme on [my branch](https://github.com/xuv/shaarli-2004licious-theme/tree/v8-bugs) to solve the PHP errors. But I can't seem to reproduce the login error that  @jasonpearce is signaling.

@jasonpearce : Could you provide more info regarding the specific login error you have ?

**Comment by jasonpearce** [comment]
@xuv Thank you for offering to help. It is appreciated.

I'm going to attempt to define my environment well, reproduce the error, and leave me site with your theme up and running for the week so you may review (perhaps longer, for I have a busy two weeks at work).

I am running Ubuntu 14 LTS. It's on Digital Ocean. I used ServerPilot.io to automatically configure the server with a LAMP stack and NGINX. I also make use of CloudFlare.

I started with **shaarli-v0.8.4-full.zip** and this **config.json.php** file (my BEFORE):

```
    "resource": {
        "data_dir": "data",
        "config": "data\/config.php",
        "datastore": "data\/datastore.php",
        "ban_file": "data\/ipbans.php",
        "updates": "data\/updates.txt",
        "log": "data\/log.txt",
        "update_check": "data\/lastupdatecheck.txt",
        "raintpl_tpl": "tpl\/",
        "raintpl_tmp": "tmp\/",
        "thumbnails_cache": "cache",
        "page_cache": "pagecache"
```

I uploaded your wonderful **2004licious** folder into the **tpl** folder and your **user.css** file into the **inc** folder. I then changed my **config.json.php** file to the following to activate your theme (my AFTER):

```
    "resource": {
        "data_dir": "data",
        "config": "data\/config.php",
        "datastore": "data\/datastore.php",
        "ban_file": "data\/ipbans.php",
        "updates": "data\/updates.txt",
        "log": "data\/log.txt",
        "update_check": "data\/lastupdatecheck.txt",
        "raintpl_tpl": "tpl\/2004licious\/",
        "raintpl_tmp": "tmp\/",
        "thumbnails_cache": "cache",
        "page_cache": "pagecache"
```

At this point, I have two known problems. 

My first (and might be related) problem is that my Home menu links to https://bookmarks.jasonpearce.com/?, which results in a "File not found." This occurs with our without your theme, so I assume it is unrelated. My assumption is that it has something to do with NGINX being in front of Apache (as ServerPilot.io configuration). I can live with this error (not a deal breaker) and documented it here: https://github.com/shaarli/Shaarli/issues/853

My second error occurs only when your theme is being used. If I logout and then attempt to log back in by clicking on Login (https://bookmarks.jasonpearce.com/?do=login), I get the following error:

`Fatal error: Uncaught ArgumentCountError: Too few arguments to function ban_canLogin(), 0 passed in /srv/users/serverpilot/apps/bookmarks/public/tmp/loginform.6414379763130b83255e7dcffb11a423.rtpl.php on line 5 and exactly 1 expected in /srv/users/serverpilot/apps/bookmarks/public/index.php:383 Stack trace: #0 /srv/users/serverpilot/apps/bookmarks/public/tmp/loginform.6414379763130b83255e7dcffb11a423.rtpl.php(5): ban_canLogin() #1 /srv/users/serverpilot/apps/bookmarks/public/inc/rain.tpl.class.php(175): include('/srv/users/serv...') #2 /srv/users/serverpilot/apps/bookmarks/public/application/PageBuilder.php(134): RainTPL->draw('loginform') #3 /srv/users/serverpilot/apps/bookmarks/public/index.php(800): PageBuilder->renderPage('loginform') #4 /srv/users/serverpilot/apps/bookmarks/public/index.php(2219): renderPage(Object(ConfigManager), Object(PluginManager)) #5 {main} thrown in /srv/users/serverpilot/apps/bookmarks/public/index.php on line 383`

If I delete everything in the tmp folder, the issue remains (but new cache files are created). 
I miss del.icio.us and am excited about the idea of maybe hosting my own homage. You've done a great job with the theme. I hope me report helps you refine and help others. Thank you.

**Comment by ArthurHoaro** [comment]
Hi, your issue is related to older version of Shaarli compatibility with PHP 7.1, see shaarli/Shaarli#714

**Comment by rlafuente** [comment]
@jasonpearce can you confirm this is the case? I've also merged @xuv's pull request that tries to deal with this (i think). Let us know so that we can either look more closely or close this issue in case you get it working.
@ArthurHoaro, great to see you here! :-)

**Comment by jasonpearce** [comment]
@ArthurHoaro Thank you for commenting. Yes. I am using PHP 7.1. I used the following example from your wiki to install Shaarali.

```
$ wget https://github.com/shaarli/Shaarli/releases/download/v0.8.4/shaarli-v0.8.4-full.zip
$ unzip shaarli-v0.8.4-full.zip
$ mv Shaarli /path/to/shaarli/
```

If I do not use the Delicious theme, almost everything works well. I can logon and logoff without error. It's not until I add the Delicious theme that I have a problem logging back in.

I think the options you are suggesting is to either a) downgrade to PHP 7.0 or something lower or b) use a version of Shaarli greater than v0.8.4. 

Tried option B (fail):
Trying option b, I first installed v0.8.4 as before. I then downloaded shaarli-latest and FTPed it on top of v0.8.4. This did not work, and introduced more problems.

Tried option A (fail):
I changed my PHP to 7.0 and then 5.6. I'm still getting more PHP errors. I think I need to spend more time with it (I'm busy and rushed, sorry).

**Comment by jasonpearce** [comment]
I reverted the server back to my prior installation. This time, all I changed was to demote PHP from 7.1 to 5.6 (option A). This time it worked. The only bug I've observed now is in Tools:

> Plugin administration: Enable, disable and configure plugins. 
> 
> Notice: Undefined index: config in /srv/users/serverpilot/apps/bookmarks/public/tmp/tools.6414379763130b83255e7dcffb11a423.rtpl.php on line 13
> Change password: Change your password. 
> 
> Rename/delete tags: Rename or delete a tag in all links 

Lesson learned, PHP version matters.

**Comment by ArthurHoaro** [comment]
> I think the options you are suggesting is to either a) downgrade to PHP 7.0 or something lower or b) use a version of Shaarli greater than v0.8.4.

No, Shaarli v0.8.4 is compatible with PHP 7.1, but this theme isn't yet. The [PR](https://github.com/shaarli/Shaarli/pull/714/files) I linked shows the issue. But you'll likely get other issue since theme variables can change a bit every time a new version is released.

**Comment by xuv** [comment]
@ArthurHoaro Thanks for this explanation.

@jasonpearce I've updated this delicious theme to work with v0.8 an PHP 7.1. You'll find it in the [v8-compat](https://github.com/xuv/shaarli-2004licious-theme/tree/v8-compat) branch in my fork. I tested it on [my own Shaarli instance](http://b.xuv.be) and it seems to work fine.

Let me know if that works for you.

**Comment by xuv** [comment]
@jasonpearce Have you had the time to test this v8-compat branch I mention? This branch also now  exists on this repo. So you don't have to pull from me anymore.

Could you please give use some feedback so we can maybe close this issue. Thank you.

**Comment by jasonpearce** [comment]
@xuv I'm sorry. You've been kind enough to assist and I have been rudely unavailable. Thank you for being patient with me.

My steps:
* Switched my subdomain to use PHP 7.1
* Downloaded a fresh version of  shaarli-v0.8.4-full.zip
* Downloaded a fresh version of shaarli-2004licious-theme-8-compat.zip
* Backed up and deleted my existing files
* Unzipped and uploaded shaarli-v0.8.4-full.zip
* Used its default theme and interactive first-run to create username and password
* Unzipped and uploaded shaarli-2004licious-theme-8-compat.zip
* Was able to successfully logout and logon

Resolved! Nice work. 

I should likely note a separate issue via a separate thread. When I went to https://bookmarks.jasonpearce.com/?do=import to import by old backups, the top of the page had these two errors:

> Notice: A non well formed numeric value encountered in /srv/users/serverpilot/apps/bookmarks/public/index.php on line 486
> 
> Notice: A non well formed numeric value encountered in /srv/users/serverpilot/apps/bookmarks/public/index.php on line 486

I was still able to import by bookmarks.html file, so the page still functioned. 

Thank you to both you and Shaarli. I love now hosting my own del.icio.us throwback subdomain on my own server. Your work is much appreciated.

**Comment by xuv** [comment]
Hello @jasonpearce 

Thanks for your feedback. Glad we could help. Indeed your bookmarks look beautiful now :)
As for the error you mention. I don't think it's related to the theme. I don't have that error when I go check the import page. I have not gone through the process of importing any file either.
I encourage you to discuss this import bug in the [Shaarli repo](/shaarli/shaarli/issues).

Have a nice day.
